### PR TITLE
[DOCS] Add documentation for new built-in roles

### PR DESCRIPTION
**Pull request by bytebilly**
Documenting https://github.com/elastic/elasticsearch/pull/71710

### CODE REVIEW

**Review by jkakavas — APPROVED** [review]
LGTM

**Review by lcawl — APPROVED** [review]
LGTM too, thanks!

**Comment by bytebilly** [comment]
@lcawl I slightly changed the notes section, could you please take a look if it's still good? Thanks!

**Comment by lcawl** [comment]
> @lcawl I slightly changed the notes section, could you please take a look if it's still good? Thanks!

I've updated the descriptions to use the built-in "experimental" text per https://github.com/elastic/docs#beta-dev-and-experimental

**Comment by bytebilly** [comment]
> > @lcawl I slightly changed the notes section, could you please take a look if it's still good? Thanks!
> 
> I've updated the descriptions to use the built-in "experimental" text per https://github.com/elastic/docs#beta-dev-and-experimental

Thanks @lcawl! I was not aware of that, good catch.

**Comment by bytebilly** [comment]
After talking with TimV, I added a more explicit note about how those roles may change in the future to adapt to new features.

@lcawl I see that there are errors (that I guess are not related), could we merge it anyway? Thanks!

File: x-pack/docs/en/security/authorization/built-in-roles.asciidoc [hunk 1]
```diff
@@ -49,6 +49,19 @@ use {transforms}. This role also includes all
...
 
+[[built-in-roles-editor]] `editor` ::
+experimental:[]
+Grants full access to all features in {kib} (including Solutions) and read-only access to data indices.
++
+--
+[NOTE]
+===============================
+* This role provides read access to any index that is not prefixed with a dot.
+* This role will automatically grant full access to new {kib} features as soon as they will be released.
```
Comment (lcawl) [inline x-pack/docs/en/security/authorization/built-in-roles.asciidoc:60]:
```suggestion
* This role automatically grants full access to new {kib} features as soon as they are released.
```

File: x-pack/docs/en/security/authorization/built-in-roles.asciidoc [hunk 2]
```diff
@@ -181,6 +194,19 @@ to the cluster state. This means users can view the metadata over all indices,
...
 
+[[built-in-roles-viewer]] `viewer` ::
+experimental:[]
+Grants read-only access to all features in {kib} (including Solutions) and to data indices.
++
+--
+[NOTE]
+===============================
+* This role provides read access to any index that is not prefixed with a dot.
+* This role will automatically grant read-only access to new {kib} features as soon as they will be available.
```
Comment (lcawl) [inline x-pack/docs/en/security/authorization/built-in-roles.asciidoc:205]:
```suggestion
* This role automatically grants read-only access to new {kib} features as soon as they are available.
```

**Comment by bytebilly** [comment]
Thanks, merging!

**Pull request merged**